packages/react/counter/src/Counter.tsx

```diff
@@ -137,33 +118,11 @@ export function Counter(props: CounterProps) {
     }
   }, [isStatic, value, query, loading])
 
-  React.useEffect(() => {
-    async function setup() {
-      if (isStatic) {
-        setDataValue(value)
-      }
-
-      if (!isStatic) {
-        const fetchedValue = await fetchData()
-
-        if (fetchedValue === undefined) {
-          setHasError(true)
-          // console.error(`QueryError: Your metric ${query?.metric} returned undefined.`) we will set logs as a feature later
-          return
-        }
-
-        setDataValue(fetchedValue)
-      }
-    }
-
-    setup()
-  }, [fetchData, isStatic, query?.metric, value])
```

Comment:
I'm pretty sure this is not necessary. Again, this is functionality @tanstack/react-query provides us.